docs: add self-hosted FIPS images guide (#3712)
langchain-ai/docs: src/langsmith/self-host-fips.mdx

---
title: FIPS-compliant images
sidebarTitle: FIPS-compliant images

> description: Run a self-hosted LangSmith on FIPS 140 compliant container images built on Chainguard FIPS.

Reviewer: ```suggestion
description: Run self-hosted LangSmith on a FIPS 140 compliant container images built on Chainguard FIPS.
```

Does that work better?

Author: should it be `on FIPS 140 compliant container images built on Chainguard FIPS`? since images is plural?

---

<Note>

>   FIPS and airgapped LangSmith deployments require a conversation with your LangChain account executive before rollout. Reach out to scope licensing, supported configurations, and upgrade paths before you change your installation.

Reviewer: ```suggestion
  FIPS and airgapped [LangSmith deployment](/langsmith/deployments) require a conversation with your LangChain account executive before rollout. Reach out to scope licensing, supported configurations, and upgrade paths before you change your installation.
```

Author: in this case its a self hosted deployment, not langsmith deployments

</Note>

As of v15, every LangChain-authored LangSmith image has a `-fips` counterpart that runs in FIPS 140 mode. Use these images when your self-hosted deployment needs FIPS compliance, for example in federal agencies, defense contractors, and regulated industries.

## How the images are built

> The `-fips` variants are built on top of [Chainguard FIPS container images](https://edu.chainguard.dev/chainguard/fips/fips-images/), which ship NIST-validated cryptographic modules (OpenSSL FIPS provider, Bouncy Castle FIPS, or BoringCrypto depending on the base). See [Chainguard's FIPS commitment](https://edu.chainguard.dev/chainguard/fips/fips-images/) for the list of modules and their CMVP certificates.

Reviewer: ```suggestion
The `-fips` variants are built on top of [Chainguard FIPS container images](https://edu.chainguard.dev/chainguard/fips/fips-images/), which ship NIST-validated cryptographic modules (OpenSSL FIPS provider, Bouncy Castle FIPS, or BoringCrypto depending on the base). For the list of modules and their CMVP certificates, refer to [Chainguard's FIPS commitment](https://edu.chainguard.dev/chainguard/fips/fips-images/).
```

Every LangChain-authored image has a `-fips` counterpart published at the same tag as the non-FIPS version:

| Non-FIPS image | FIPS image |
| --- | --- |
| `langchain/langsmith-ace-backend` | `langchain/langsmith-ace-backend-fips` |
| `langchain/langsmith-backend` | `langchain/langsmith-backend-fips` |
| `langchain/langsmith-frontend` | `langchain/langsmith-frontend-fips` |
| `langchain/langsmith-go-backend` | `langchain/langsmith-go-backend-fips` |
| `langchain/langsmith-playground` | `langchain/langsmith-playground-fips` |
| `langchain/hosted-langserve-backend` | `langchain/hosted-langserve-backend-fips` |
| `langchain/langgraph-operator` | `langchain/langgraph-operator-fips` |

PostgreSQL, Redis, and ClickHouse are not published as FIPS variants by LangChain. If your deployment requires FIPS for these components, bring your own FIPS-mode service and connect via [external Postgres](/langsmith/self-host-external-postgres), [external Redis](/langsmith/self-host-external-redis), or [external ClickHouse](/langsmith/self-host-external-clickhouse).

## Use FIPS images

> Update `values.yaml` in your LangSmith Helm installation to point each LangChain image repository at its `-fips` counterpart, keeping your existing tag. Replace `0.15.0` with the LangSmith version you want to deploy:

Reviewer: ```suggestion
Update `values.yaml` in your LangSmith Helm installation to point each LangChain image repository at its `-fips` counterpart, keeping your existing tag. Replace `0.15.0` with the [LangSmith version](/langsmith/self-hosted-changelog) you want to deploy:
```

```yaml
images:
  aceBackendImage:
    repository: "langchain/langsmith-ace-backend-fips"
    pullPolicy: IfNotPresent
    tag: "0.15.0"
  backendImage:
    repository: "langchain/langsmith-backend-fips"
    pullPolicy: IfNotPresent
    tag: "0.15.0"
  frontendImage:
    repository: "langchain/langsmith-frontend-fips"
    pullPolicy: IfNotPresent
    tag: "0.15.0"
  hostBackendImage:
    repository: "langchain/hosted-langserve-backend-fips"
    pullPolicy: IfNotPresent
    tag: "0.15.0"
  operatorImage:
    repository: "langchain/langgraph-operator-fips"
    pullPolicy: IfNotPresent
    tag: "0.15.0"
  platformBackendImage:
    repository: "langchain/langsmith-go-backend-fips"
    pullPolicy: IfNotPresent
    tag: "0.15.0"
  playgroundImage:
    repository: "langchain/langsmith-playground-fips"
    pullPolicy: IfNotPresent
    tag: "0.15.0"
```

> Apply the change and upgrade following [Upgrading LangSmith](/langsmith/self-host-upgrades).

Reviewer: ```suggestion
Apply the change and upgrade following the [Upgrading LangSmith](/langsmith/self-host-upgrades) guide.
```
